lib/spire/order.rb

```diff
@@ -1,94 +1,218 @@
+        :created_by, :modified_by, :created, :modified, :order_no
       ]
 
-    validates_presence_of :customer, :items
+    validates_acceptance_of :id, :customer, :address, :shipping_address, :items
+
+    ACTIVE = "0"
+    ON_HOLD = "1"
+    INACTIVE = "2"
 
     SYMBOL_TO_STRING = {
       id: 'id',
-      background_color: 'backgroundColor', 
+      order_no: 'orderNo',
       customer: 'customer',
+      status: 'status',
+      type: 'type',
+      hold: 'hold',
+      order_date: 'orderDate',
       address: 'address',
-      shipping_address: "shippingAddress",
+      shipping_address: 'shippingAddress',
+      customer_po: 'shippingPO',
+      fob: 'fob',
+      terms_code: "termsCode",
+      terms_text: 'termsText',
+      freight: 'freight',
+      taxes: 'taxes',
+      subtotal: 'subtotal',
+      subtotal_ordered: 'subtotalOrdered',
+      discount: 'discount',
+      total_discount: 'totalDiscount',
+      total: 'total',
+      total_ordered: 'totalOrdered',
+      gross_profit: 'grossProfit',
       items: 'items',
       created_by: 'createdBy',
-      modified_by: 'modifiedBy'
+      modified_by: 'modifiedBy',
+      created: 'created',
+      modified: 'modified',
+      background_color: 'backgroundColor'
     }
 
     class << self
-      # Find a specific order by its id
+       # Find a specific order by its id.
       #
       # @raise [Spire::Error] if the order could not be found.
       #
       # @return [Spire::Order]
-      def find(id)
-        client.find('/sales/orders', id)
+
+      def find(id, params = {})
+        client.find('/sales/orders', id, params)
       end
 
-      # Search for orders by query.
+      # Search for order by query. This will even return inactive orders!
       #
-      # @raise [Spire::Error] if the item could not be found.
+      # @raise [Spire::Error] if the order could not be found.
       #
       # @return [Spire::Order]
+
       def search(query)
         client.find_many(Spire::Order, '/sales/orders/', {q: query})
       end
 
-      # Create a new order and save it on Spire.
+      # Create a new item and save it on Spire.
       #
       # @param [Hash] options
-      #
-      # @raise [Spire::Error] if the item could not be created.
+      # @option options [Hash] :customer the spire customer who will be associated with the order
+      # @option options [Hash] :address this is the billing address for the customer
+      # @option options [Hash] :shipping_address this is the shipping address that the order will be sent to (defaults to the billing address if none provided)
+      # @option options [Array] :items this is an array of hashes that will accept an inventory item that will have a hash example input from the web client: items: [ { "inventory": {"id": 123} } ]
+      # this array can also accept a desctiption and comment object that will create a comment on the order itself ex: items: [{"description":"MAKE COMMENT THRU API","comment":"MAKE COMMENT THRU API"}]
```

Comment:
```suggestion
      # this array can also accept description and comment objects that will create a comment on the order itself ex: items: [{"description":"MAKE COMMENT THRU API","comment":"MAKE COMMENT THRU API"}]
```

Reply:
this is singular as it is one description and one comment object rather than a collection